config/config-server/test/com/thoughtworks/go/config/MagicalGoConfigXmlWriterTest.java

```diff
@@ -339,6 +339,53 @@ public void shouldWriteEmptyOnCancelTaskWhenDefined() throws Exception {
     }
 
     @Test
+    public void shouldNotWriteMoreThanOneOnCancelTaskWhenDefined() throws Exception {
+        String xml = "<cruise schemaVersion='30'>\n"
+                + "<server artifactsdir='artifactsDir' >"
+                + "<mailhost hostname=\"10.18.3.171\" port=\"25\" username=\"cruise2\" password=\"password\" tls=\"false\" from=\"cruise2@cruise.com\" admin=\"ps@somewhere.com\" />"
+                + "<security>"
+                + "<ldap uri=\"ldap://blah.blah.somewhere.com\" managerDn=\"cn=Active Directory Ldap User,ou=SomeSystems,ou=Accounts,ou=Principal,dc=corp,dc=somecompany,dc=com\" managerPassword=\"password\" searchBase=\"ou=Employees,ou=Company,ou=Principal,dc=corp,dc=somecompany,dc=com\" searchFilter=\"(sAMAccountName={0})\" />"
+                + "</security>"
+                + "</server>"
```

Comment:
I think the mailhost and security tags are not mandatory. exclude them fit this test.